src/Domain/PriceMap/PriceMapProductPrice.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Domain\PriceMap;
+
+use App\Models\Model;
+use App\Models\Product;
+use Brick\Money\Money;
+use Domain\Currency\Currency;
+use Domain\Price\Dtos\PriceDto;
+use Illuminate\Database\Eloquent\Casts\Attribute;
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Relations\BelongsTo;
+use Spatie\LaravelData\WithData;
+
+/**
+ * @mixin IdeHelperPriceMapProductPrice
+ */
+final class PriceMapProductPrice extends Model
```

Comment:
`PriceMapProductPrice` and `PriceMapSchemaOptionPrice` differ only in their relationship with the product/option. It might be better to move the duplicated code to a Trait or a separate class.